src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,142 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private static StorageDao storageDao;
+    private User correctUser;
+    private User wrongUser;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        correctUser = new User();
+        correctUser.setAge(23);
+        correctUser.setLogin("david23");
+        correctUser.setPassword("1d2f3h");
+    }
+
+    @Test
+    void register_userNull_notOk() {
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(null);
+        });
+    }
+
+    @Test
+    void register_Ok() {
+        registrationService.register(correctUser);
+        assertEquals(storageDao.get(correctUser.getLogin()), correctUser);
+    }
+
+    @Test
+    void register_loginDuplicate_notOk() {
+        storageDao.add(correctUser);
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(correctUser);
+        });
+    }
+
+    @Test
+    void register_age_notOk() {
+        correctUser.setAge(17);
+        try {
+            registrationService.register(correctUser);
+        } catch (RuntimeException e) {
+            return;
+        }
+        fail("Runtime exception must be thrown if age less than 18!");
```

Comment:
better don't use try-catch construction to handle an exception, we already have static method in jUnit library for this )